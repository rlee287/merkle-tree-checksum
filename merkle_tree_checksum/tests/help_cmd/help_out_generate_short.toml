bin.name = "merkle_tree_checksum"
args = "generate-hash -h"
status.code = 1
