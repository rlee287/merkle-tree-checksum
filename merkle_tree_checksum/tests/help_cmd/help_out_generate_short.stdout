Generates Merkle tree hashes

Usage: merkle_tree_checksum generate-hash [OPTIONS] --output <output> -- <FILES>...

Arguments:
  <FILES>...  Files to hash

Options:
  -f, --hash-function <hash>      Hash function to use [default: sha256] [possible values: crc32,
                                  sha224, sha256, sha384, sha512, sha512_224, sha512_256, sha3_224,
                                  sha3_256, sha3_384, sha3_512, blake2b512, blake2s256, blake3]
  -b, --branch-factor <branch>    Branch factor for tree [default: 4]
  -l, --block-length <blocksize>  Block size to hash over, in bytes [default: 4096]
  -o, --output <output>           Output file
      --overwrite                 Overwrite output file if it already exists
  -s, --short                     Write only the summary hash
  -h, --help                      Print help (see more with '--help')

Supported hash functions are the SHA2 family, the SHA3 family, Blake2b/Blake2s, Blake3, and CRC32.
sha512-based hashes (sha384, sha512, sha512_224, and sha512_256) can be significantly faster than
sha256-based hashes (sha224 and sha256) on 64-bit systems that lack SHA hardware acceleration.
